README.md
# Tactics-Remix

[![Latest Release](https://img.shields.io/github/v/release/gibberlings3/Tactics-Remix?include_prereleases)](https://github.com/Gibberlings3/Tactics-Remix/releases/latest)
[![Github downloads (all releases)](https://img.shields.io/github/downloads/gibberlings3/Tactics-Remix/total.svg?color=informational)](https://github.com/gibberlings3/Tactics-Remix/releases)
![Language](https://img.shields.io/static/v1?label=language&message=english%20%7C%20french&color=informational)
![Platform](https://img.shields.io/static/v1?label=platform&message=windows%20%7C%20macos%20%7C%20linux&color=informational)

[A Gibberlings Three Mod](https://www.gibberlings3.net/)

**Author**: [morpheus562](https://www.gibberlings3.net/profile/11591-morpheus562/)

**On the Web**: [Home Page](https://www.gibberlings3.net/mods/quests/tactics-remix/) and [Discussion Topic](https://www.gibberlings3.net/forums/topic/35950-tactics-remix/)

## Introduction

It is with great pleasure I have been given permission from Wes to update his original [Tactics mod](https://weidu.org/tactics.html) to be compatible with the Enhanced Editions. Tactics Remix adds quite a few optional, but difficult, encounters to BG2:EE and EET aims to make the game more challenging and rewarding.

Many of these components have been drawn from different mod authors; I did not write everything here. In particular, Wes Weimer, Gebhard Blucher, Kensai Ryu, Xyx and Ishan have made contributions. I have gone through, updated the components for the EE's, and redid most of the AI from the ground up. The goal is for the encounters to feel similar to their oBG counterparts with more challenging enemy AI.

## Compatibility

If you are upgrading from an earlier version of this mod, I recommend uninstalling all mod components and replacing all files with the new ones before doing a fresh install. If you choose to use SCS, parts of this mod should be installed before it while other parts should be installed after. These will be clearly outlined during the install process.

These scripts are currently written for a vanilla/SCS understanding of the spells/abilities. Tactics Remix does not take into account other mods that alter existing spells. Some mods do lighter edits that shouldn't impact the scripts, but others, such as Spell Revisions, do a complete overhaul and will cause issues, inconsistencies, and undesired results. As such, I would not recommend install of Tactics Remix if you are playing with Spell Revisions. This may change in the future, but for now I would list these as incompatible.

## Content

Tactics Remix adds quite a few new components, and all of the components are individually optional: you may install some but not others, your choice. I have also broken up the mods into two groups: 1) Add New Encounters and 2) Update Existing Encounters. New encounters should be installed prior to SCS while updates to existing encounters should be installed after SCS (if you, the player, chooses to use SCS).

All of the enemy scripts have been rewritten from scratch and many of the fight mechanics are improved from the old Tactics days. These fights are aimed at being more challenging than the original encounters while removing much of the cheese that players of old remember. These are meant to be BRUTAL. The difficulty slider will alter the overall challenge (e.g. removing undroppable items from enemies to make them easier, less or different enemy spawns, altered enemy prebuffs, etc.). I aim to have a bit of randomness added to the encounters and the scripts are not optimized to always cast the "best" spell. Prebuffs and contingencies/sequencers will vary. Additionally, casters will randomly choose if they will be casting an offensive or a defensive/disabling spell. From there, the enemy scripts will cycle through available spells in the specific category with yet more randomness. Best spells will pop up first, but it is random if the script will choose to cast or pass. This should add a bit of variance to the encounters from reload to reload and hopefully the player enjoys the spontaneity. 

### Add New Encounters - Install BEFORE SCS

#### "Kuroisan", the Acid Kensai

Kuroisan, a Kensai / Wu Jen from Kara-Tur will approach you if you are carrying Celestial Fury or Malakar, regardless if equipped or not. Once you acquire either of these weapons, he will begin the hunt to find you (1 day) and attempt to take yours. Depending on the difficulty level, he may bring additional sword allies to join the fight. 

#### The Ritual

The Ritual component adds a new tactical quest started by speaking with the yellow-robed individual outside of Watcher's Keep. This quest takes the player through some outdoor combat and obscure symbolism against a very pretty backdrop. 

PLEASE NOTE: Original recommendation for party level is around 2 million XP. 

#### "Red Badge" Poison-Based Encounter

This challenge is triggered if you have more than 3 million XP, have either killed Gromnir or be in SoA, and are standing outside of Watcher's Keep. The "Red Badge" will wisk you away with promises to come. Once you have over 3.5 million XP with the same conditions as above, they will bring you back for your confrontation.

#### Gebhard Blucher's Lich in the Docks

A new lich can be found in the Docks district with an endless supply of undead minions to call upon. You may find her by entering the door to the north of the main Shadow Thief guildhouse entrance. 

WARNING: Reported to be *incompatible* with Back to Brynnlaw. 

#### Kensai Ryu's Gnome Fighter/Illusionist in the Docks

A gnome fighter/illusionist full of tricks is looking to collect upon the bounty on Valygar's head. Beware on your search as the gnome will kill anyone in his way. 

### Update Existing Encounters - Install AFTER SCS

These components will check if SCS is present and update accordingly to prevent any compatibility issues.

#### Improved Sahuagin City

This component makes all Sahuagin fiercer opponents. It also updates the fight against the Priestess at the heart of the City of Caverns who will now summon allies to come to her aid. Beware, on Insane Difficulty she will summon the Minor Avatar of Sekolah.

#### Morpheus562's Improved Planar Prison Bounty Hunters

This component makes the fight against the Planar Prison bounty hunters a bit more difficult. Aawill the Bard is now a Blade and will use spells commensurate with his level. His companions will also be better equipped, and an additional companion may join the fight at higher difficulties.

#### Improved Irenicus

I take the original Tactics fight, with some slight alterations, to give Joneleth "The Shattered One" Irenicus the fight he and the player truly deserve. This is a multi-phase fight that should provide a suitable challenge. To start, instead of summoning generic demons, Jon will shatter himself into multiple fragments and brings forth enemies for each Tear of Bhaal Test in Hell. These enemies will provide their own unique challenges, and will bring forth more minions to aid themselves, depending on the difficulty. 

#### Improved Guarded Compound in the Temple District

This component provides a drastic increase in difficulty for the guardians of Celestial Fury. Enemies will have appropriate gear and spell picks for their levels. Enemies will prepare for your arrival and prebuff, and once you arrive there is no escape to the lower floor until you are victorious.

PLEASE NOTE: This component pairs well with the Forgotten Armament component to add net new items into the game for yet an additional difficulty increase. 

#### Improved Twisted Rune

Enemies in this encounter will behave as characters of their level should with better decision making and gear. Original Tactics controls to reduce player cheese remain in place. Revanek, instead of being naked, will become Wizard Slayer decked out in new gear with this update. To spice things up, I am assuming the upgrades from the Skills and Abilities Wizard Slayer are or will be installed. This means, if EEex is present, Revanek's hits will have a 10% chance of bypassing PfMW and the like and dispelling the target.

#### Gebhard Blucher's Improved Demon Knights

This component makes the Demon Knights at the Demogorgon altar in the Underdark overall better opponents and more of a challenge to the player. 

#### Kensai Ryu's Tougher Kangaxx and Guardians

The tomb of Kangaxx is protected by a new set of guardians above. They are looking to acquire his ring and will kill any they perceive as getting in the way. If SCS is installed, Kangaxx will revert to SCS AI; however, the guardians will still remain.

#### Kensai Ryu's Improved Crypt King

The Crypt King is now more worthy of that name and will summon more undead minions every now and then. I have overhauled the fight mechanics to make it even more challenging than the original Tactics version.

#### Smarter Dragons (EARLY TESTING - COMING SOON)

As Wes stated: "Dragons clearly got the short end of the stick somewhere. They're just not awe-inspiring in BGII. In fact, they're pretty much pushovers, especially with half-decent tactics on your part." We are not playing Dungeons and Demiliches, or Dungeons and Fire Giants, or Dungeons and Mindflayers. The aim of this component is to vasty improve the power and challenge when facing a dragon.

#### Kensai Ryu's Improved Copper Coronet

Simply adds more guards in the Copper Coronet when they go hostile. 

#### Mike Barnes' Improved Small Teeth Pass

Adds three dragons to Small Teeth Pass (AR1700).

WARNING: Reported to be *incompatible* with TDD. 

#### Morpheus562's Improved Drizzt (BGEE)

This makes the BGEE version of Drizzt more formidable. If he goes hostile, he will prioritize attack the player instead of the gnolls. 

#### Morpheus562's Improved Drizzt (BG2EE)

This makes the BG2EE version of Drizzt and his companions more formidable. Additionally, if playing on Insane difficulty or higher he will bring in a familiar mage to aid him. Should the mage be defeated, any gear acquired from the battle will not be taken from the party at a later time. 

#### Tougher Fire Giants

Damage output and difficulty for fire giants is greatly increased.

#### Morpheus562's Tougher Githyanki

Githyanki will now be better equipped, better leveled, and overall pose more of a challenge to the player. 

#### Morpheus562's Tougher Yuan-ti

Yuan-ti will receive a number of immunities, to include some magic resistance (40%) and complete resistance to poison. Additionally, they receive some snake like enhancements to include infravision, faster casting speed, faster weapon speed, and an extra 0.5 apr. Yuan-ti fighters will now be wielding magical scimitars and leather armors.

PLEASE NOTE: With my Skills and Abilities Proficiency overhaul and Forgotten Armament's NWN Style Weapons and Updates to Existing Weapons will greatly increase the deadliness of Yuan-ti fighters. The average Yuan-ti fighter and Greater Yuan-ti will have a crit range of 16-20, and Yuan-ti Elites will have a crit range of 15-20. 

#### More encounters will be added.

### Enemy Improvements - Install AFTER SCS

#### Improve All Enemies Hit Points

There are a couple options for the player to select to increase the HP of all enemies. There is an additional option to adjust enemy HP based on the ini file if the player does not like the pre-defined options. Friendly cast summons will NOT have their HP altered. 

## Special Thanks

- Wes thank you for your permission to update such an amazing mod! I appreciate being a steward of your work, and I hope you find it enjoyable.
- Gebhard Blucher, Kensai Ryu, Xyx, Mike Barnes, and Ishan, thank you for all your amazing work on the original Tactics mod. I hope you too enjoy the work I have put into it.
- Cam for all of your help and assistance. The reasons are many, thank you.
- JohnBob for providing the French translation.
- alaisFcZ for providing the Spanish translation.
- All those who assisted in testing and providing feedback! I appreciate every single one of you for all that you do!

## Further Reading

- [Installation](INSTALL.md)
- [Change Log](CHANGELOG.md)


CHANGELOG.md
# Changelog

## Version 0.7-beta

November xx, 2023

### Changed

- Added Tougher Dragons component.
- Updated Lich in the Docks to not overwrite the area script if one already exists.

## Version 0.6.1-beta

September 26, 2023

### Changed

- Spanish translation courtesy of alaisFcZ.
- Slight correction to Improved Fire Giants if MiH - Item Pack is installed.
- Twisted Rune difficulty is now better tied to the difficulty slider.
- Kangaxx guardians difficulty is now better tied to the difficulty slider.

## Version 0.6-beta

September 20, 2023

### Changed

- Added Improved Sahuagin City component.
- Added Morpheus562's Improved Planar Prison Bounty Hunters component.
- Added Morpheus562's Tougher Githyanki component.
- Added Morpheus562's Tougher Yuan-ti component.
- Added Morpheus562's Improved Drizzt (BG2EE) component.
- Kuroisan rewards tied to the difficulty slider.

## Version 0.5.1-alpha

August 17, 2023

### Changed

- Corrected bug with Kuroison script on creation.
- Added Tougher Fire Giants component.

## Version 0.5-alpha

August 16, 2023

### Changed

- Nerfed Shatter Magic to have chance of dispel instead of always dispel.
- Cam provided updates to MOFAM01.cre and MOFAM02.cre.
- Vastly increased the difficulty for Kuroisan and Guarded Compound encounters.

## Version 0.4.1-alpha

August 3, 2023

### Changed

- Forgot to push changes...

## Version 0.4-alpha

August 3, 2023

### Changed

- Removed exploit against the Lich in the Docks.
- Corrected a typo where the Yellow Dragon Armor had the wrong name in the description.
- Updated Kuroisan compatibility with Tweaks Anthology 2H Katanas component.
- Updated Resilient Trodders to be available to all classes.
- Updated text in Greater Robe of Eyes.
- Corrected bug with wrong name for Uthgardt Holy Shield would display.
- Corrected Lich in the Docks to only have 2 uses of Timestop.
- Corrected bug in Improved Irenicus preventing the fight to progress to phase 2.

## Version 0.3-alpha

June 5, 2023

### Changed 

- Added Improved Irenicus.
- Added Kensai Ryu's Improved Copper Coronet.
- Added Mike Barnes' Improved Small Teeth Pass.
- Updated Revanek to have Wizard Slayer abilities.

## Version 0.2.1-alpha

April 6, 2023

### Changed 

- Corrected file paths for Kuroisan.

## Version 0.2-alpha

March 31, 2023

### Changed 

- Added Gebhard Blucher's Improved Demon Knights.
- Added Kensai Ryu's Tougher Kangaxx and Guardians.
- Added Morpheus562's Improved Drizzt (BGEE).
- Added component to increase all enemy hit points.
- Removed bug from Lich in the Docks that continued to spawn enemies after the lich was defeated.
- Cam updates to items.
- Remove cheese from Fallowmist.
- Corrected bug where I accidentally used the same item file name across mods.
- Added French translation courtesy of JohnBob.
- Organized baf file structure.

## Version 0.1-alpha

September 22, 2022

### Changed 

- Initial creation and alpha release.


tactics-remix/dlg/ii5.d
// Irenicus The Slayer

BEGIN ii5 


IF ~NumTimesTalkedTo(0)~ THEN BEGIN s0
  SAY @517
  IF ~~ THEN REPLY @518 GOTO s1
  IF ~~ THEN REPLY @519 GOTO s1
END

/*
IF ~~ THEN BEGIN s1
  SAY @520 = @521 = @522
  IF ~~ THEN DO ~ActionOverride(Player1,DropInventory())~ GOTO s2
END

IF ~~ THEN BEGIN s2
  SAY @523
  IF ~~ THEN DO 
          ~CreateVisualEffectObject("SPPLANAR",Player1)
           ActionOverride(Player1,JumpToPoint([1900.2100])) 
           Enemy()~ EXIT
END
*/

IF ~~ THEN BEGIN s1
  SAY @520 = @521 = @523
  IF ~~ THEN DO
          ~CreateVisualEffectObject("SPPLANAR",Player1)
           ActionOverride(Player1,JumpToPoint([1900.2100]))
           MoveViewPoint([1900.2100],INSTANT) 
           Enemy()~ EXIT
END

// Irenicus the Shattered One
BEGIN ii6

IF ~NumTimesTalkedTo(0)~ THEN BEGIN s0
  SAY @524
  IF ~~ THEN REPLY @525 GOTO s1
  IF ~~ THEN REPLY @526 GOTO s1
  IF ~~ THEN REPLY @527 GOTO s1
END

IF ~~ THEN BEGIN s1
  SAY @528 = @529
  IF ~~ THEN DO ~CreateVisualEffectObject("SPPLANAR",Myself)
           CreateCreature("moshad01",[1900.2100],0)
           JumpToPoint([1900.2100])~ EXIT
END


tactics-remix/dlg/solak1.d
///////////////////////////////////////////////////////////////////
// Dialogue for Kuroisan, Acidid Kensai
// He wants Celestial Fury or Malakar. 
///////////////////////////////////////////////////////////////////
BEGIN SOLAK1

IF ~Global("KuroiAlready","GLOBAL",1)~ THEN BEGIN again
  SAY @120
  IF ~~ THEN DO ~Enemy()~ EXIT
END

IF ~NumTimesTalkedTo(0)~ THEN BEGIN 0
  SAY @101
  IF ~~ THEN REPLY @102 GOTO 2 
  IF ~~ THEN REPLY @103 GOTO 2
  IF ~~ THEN REPLY @104 GOTO 1
END

IF ~~ THEN BEGIN 1
  SAY @106 
  IF ~~ THEN GOTO 2
END

IF ~~ THEN BEGIN 2
  SAY @105 = @107
  IF ~!PartyHasItem("sw1h51")
!PartyHasItem("sw1h45") /* Malakar */
!PartyHasItem("c2sw1h02") /* Celestial Fury +5 */
!PartyHasItem("cdkt4") /* Celestial Fury +5 */
!PartyHasItem("c!kt4")~ /* Celestial Fury +5 */
    THEN GOTO neither
  IF ~PartyHasItem("c2sw1h02")~ /* Celestial Fury +5 */
    THEN GOTO fury5
  IF ~PartyHasItem("cdkt4")~ /* Celestial Fury +5 */
    THEN GOTO fury51
  IF ~PartyHasItem("c!kt4")~ /* Celestial Fury +5 */
    THEN GOTO fury52
  IF ~PartyHasItem("sw1h51")~ /* Celestial Fury */
    THEN GOTO fury
  IF ~PartyHasItem("cdkt23")~ /* Celestial Fury */
    THEN GOTO fury1
  IF ~PartyHasItem("c!kt23")~ /* Celestial Fury */
    THEN GOTO fury2
  IF ~!PartyHasItem("sw1h51")
!PartyHasItem("cdkt23")
!PartyHasItem("c!kt23")
PartyHasItem("sw1h45")~ /* Malakar */
    THEN GOTO malakar
  IF ~!PartyHasItem("sw1h51")
!PartyHasItem("cdkt23")
!PartyHasItem("c!kt23")
PartyHasItem("cdkt22")~ /* Malakar */
    THEN GOTO malakar1
  IF ~!PartyHasItem("sw1h51")
!PartyHasItem("cdkt23")
!PartyHasItem("c!kt23")
PartyHasItem("c!kt22")~ /* Malakar */
    THEN GOTO malakar2
END

IF ~~ THEN BEGIN neither
  SAY @121
  IF ~~ THEN DO ~SetGlobal("KuroiSpawn","GLOBAL",0) 
  ReallyForceSpell(Myself,DRYAD_TELEPORT)~ EXIT
END

IF ~~ THEN BEGIN fury5
  SAY @108
  IF ~~ THEN REPLY @109 DO ~TakePartyItem("c2sw1h02")~ GOTO byebye
  IF ~~ THEN REPLY @110 GOTO fight
  IF ~~ THEN REPLY @114 GOTO whyrob
END

IF ~~ THEN BEGIN fury51
  SAY @108
  IF ~~ THEN REPLY @109 DO ~TakePartyItem("cdkt4")~ GOTO byebye
  IF ~~ THEN REPLY @110 GOTO fight
  IF ~~ THEN REPLY @114 GOTO whyrob
END

IF ~~ THEN BEGIN fury52
  SAY @108
  IF ~~ THEN REPLY @109 DO ~TakePartyItem("c!kt4")~ GOTO byebye
  IF ~~ THEN REPLY @110 GOTO fight
  IF ~~ THEN REPLY @114 GOTO whyrob
END

IF ~~ THEN BEGIN fury
  SAY @108
  IF ~~ THEN REPLY @109 DO ~TakePartyItem("sw1h51")~ GOTO byebye
  IF ~~ THEN REPLY @110 GOTO fight
  IF ~~ THEN REPLY @114 GOTO whyrob
END

IF ~~ THEN BEGIN fury1
  SAY @108
  IF ~~ THEN REPLY @109 DO ~TakePartyItem("cdkt23")~ GOTO byebye
  IF ~~ THEN REPLY @110 GOTO fight
  IF ~~ THEN REPLY @114 GOTO whyrob
END

IF ~~ THEN BEGIN fury2
  SAY @108
  IF ~~ THEN REPLY @109 DO ~TakePartyItem("c!kt23")~ GOTO byebye
  IF ~~ THEN REPLY @110 GOTO fight
  IF ~~ THEN REPLY @114 GOTO whyrob
END

IF ~~ THEN BEGIN malakar
  SAY @111
  IF ~~ THEN REPLY @109 DO ~TakePartyItem("sw1h45")~ GOTO byebye
  IF ~~ THEN REPLY @110 GOTO fight
  IF ~~ THEN REPLY @114 GOTO whyrob
END

IF ~~ THEN BEGIN malakar1
  SAY @111
  IF ~~ THEN REPLY @109 DO ~TakePartyItem("cdkt22")~ GOTO byebye
  IF ~~ THEN REPLY @110 GOTO fight
  IF ~~ THEN REPLY @114 GOTO whyrob
END

IF ~~ THEN BEGIN malakar2
  SAY @111
  IF ~~ THEN REPLY @109 DO ~TakePartyItem("c!kt22")~ GOTO byebye
  IF ~~ THEN REPLY @110 GOTO fight
  IF ~~ THEN REPLY @114 GOTO whyrob
END

IF ~~ THEN BEGIN fight
  SAY @112
  IF ~~ THEN DO ~Enemy()~ EXIT
END

IF ~~ THEN BEGIN byebye
  SAY @113
  IF ~~ THEN DO ~ReallyForceSpell(Myself,DRYAD_TELEPORT)~ EXIT
END

IF ~~ THEN BEGIN whyrob
  SAY @115 = @116 = @117 = @118
  IF ~~ THEN REPLY @110 GOTO fight
  IF ~PartyHasItem("c2sw1h02")~ /* CF+5 */
    THEN REPLY @109 DO ~TakePartyItem("c2sw1h02")~ GOTO byebye
  IF ~PartyHasItem("cdkt4")~ /* CF+5 */
    THEN REPLY @109 DO ~TakePartyItem("cdkt4")~ GOTO byebye
  IF ~PartyHasItem("c!kt4")~ /* CF+5 */
    THEN REPLY @109 DO ~TakePartyItem("c!kt4")~ GOTO byebye
  IF ~PartyHasItem("sw1h51")~ /* Celestial Fury */
    THEN REPLY @109 DO ~TakePartyItem("sw1h51")~ GOTO byebye
  IF ~PartyHasItem("cdkt23")~ /* Celestial Fury */
    THEN REPLY @109 DO ~TakePartyItem("cdkt23")~ GOTO byebye
  IF ~PartyHasItem("c!kt23")~ /* Celestial Fury */
    THEN REPLY @109 DO ~TakePartyItem("c!kt23")~ GOTO byebye
  IF ~!PartyHasItem("sw1h51")
!PartyHasItem("cdkt23")
!PartyHasItem("c!kt23")
PartyHasItem("sw1h45")~ /* Malakar */
    THEN REPLY @109 DO ~TakePartyItem("sw1h45")~ GOTO byebye
  IF ~!PartyHasItem("sw1h51")
!PartyHasItem("cdkt23")
!PartyHasItem("c!kt23")
PartyHasItem("cdkt22")~ /* Malakar */
    THEN REPLY @109 DO ~TakePartyItem("cdkt22")~ GOTO byebye
  IF ~!PartyHasItem("sw1h51")
!PartyHasItem("cdkt23")
!PartyHasItem("c!kt23")
PartyHasItem("c!kt22")~ /* Malakar */
    THEN REPLY @109 DO ~TakePartyItem("c!kt22")~ GOTO byebye
END


tactics-remix/dlg/ritpont.d
// Pontifex, the guy who gives you the ritual quest

BEGIN RITPONT

IF ~AreaCheck("AR0086")~ THEN BEGIN arrived
  SAY @120
  IF ~~ THEN DO ~GiveItemCreate("Potn11",Player1,5,5,5)
GiveItemCreate("Potn42",Player1,5,5,5)
GiveItemCreate("Restore",Player1,5,5,5)
GiveItemCreate("scrl7o",Player1,5,5,5)
GiveItemCreate("scrl7q",Player1,5,5,5)
GiveItemCreate("scrl94",Player1,5,5,5)
GiveItemCreate("scrl8w",Player1,5,5,5)
GiveItemCreate("wand10",Player1,8,8,8)
GiveItemCreate("mysboot",Player1,1,1,1)
GivePartyAllEquipment()
ReallyForceSpell(Myself,DRYAD_TELEPORT)~ EXIT
END

IF ~Global("Fight","LOCALS",1)~ THEN BEGIN fight
  SAY @210 = @211 = @212
  IF ~~ THEN DO ~Enemy()~ EXIT
END

IF ~Global("RitPrep","GLOBAL",1)~ THEN BEGIN rit
  SAY @200
  /* decision tree ahoy! */
  IF ~~ THEN GOTO perfect
  IF ~Dead("ritcount")~ THEN GOTO dead_disrupt
  IF ~Global("RitualDisrupted","GLOBAL",0)~ THEN GOTO dead_nodisrupt
  IF ~Global("RitualGuardPassed","GLOBAL",0)~ THEN GOTO never_got_in
END

IF ~~ THEN BEGIN never_got_in
  SAY @201 = @202 
  IF ~~ THEN DO ~GiveItemCreate("ritamul",Player1,1,0,0)
GiveItemCreate("ritamul",Player2,1,0,0)
GiveItemCreate("ritamul",Player3,1,0,0)
GiveItemCreate("ritamul",Player4,1,0,0)
GiveItemCreate("ritamul",Player5,1,0,0)
GiveItemCreate("ritamul",Player6,1,0,0)
Enemy()~ EXIT
END

IF ~~ THEN BEGIN dead_nodisrupt
  SAY @203 = @204
  IF ~~ THEN DO ~GiveItemCreate("ritamul",Player1,1,0,0)
GiveItemCreate("ritamul",Player2,1,0,0)
GiveItemCreate("ritamul",Player3,1,0,0)
GiveItemCreate("ritamul",Player4,1,0,0)
GiveItemCreate("ritamul",Player5,1,0,0)
GiveItemCreate("ritamul",Player6,1,0,0)
SetGlobal("Fight","LOCALS",1)
AddXPObject(Player1,30000) 
AddXPObject(Player2,30000) 
AddXPObject(Player3,30000) 
AddXPObject(Player4,30000) 
AddXPObject(Player5,30000) 
AddXPObject(Player6,30000)~ EXIT
END

IF ~~ THEN BEGIN dead_disrupt
  SAY @205 = @204
  IF ~~ THEN DO ~GiveItemCreate("ritamul",Player1,1,0,0)
GiveItemCreate("ritamul",Player2,1,0,0)
GiveItemCreate("ritamul",Player3,1,0,0)
GiveItemCreate("ritamul",Player4,1,0,0)
GiveItemCreate("ritamul",Player5,1,0,0)
GiveItemCreate("ritamul",Player6,1,0,0)
SetGlobal("Fight","LOCALS",1)
AddXPObject(Player1,30000) 
AddXPObject(Player2,30000) 
AddXPObject(Player3,30000) 
AddXPObject(Player4,30000) 
AddXPObject(Player5,30000) 
AddXPObject(Player6,30000)~ EXIT
END

IF ~~ THEN BEGIN perfect
  SAY @206 = @207
  IF ~~ THEN DO ~GiveItemCreate("ritamul",Player1,1,0,0)
GiveItemCreate("ritamul",Player2,1,0,0)
GiveItemCreate("ritamul",Player3,1,0,0)
GiveItemCreate("ritamul",Player4,1,0,0)
GiveItemCreate("ritamul",Player5,1,0,0)
GiveItemCreate("ritamul",Player6,1,0,0)
GiveItemCreate("ritdrag",Player1,1,0,0)
AddXPObject(Player1,90000) 
AddXPObject(Player2,90000) 
AddXPObject(Player3,90000) 
AddXPObject(Player4,90000) 
AddXPObject(Player5,90000) 
AddXPObject(Player6,90000)
ReallyForceSpell(Myself,DRYAD_TELEPORT)~ EXIT
END

IF ~NumTimesTalkedTo(0)~ THEN BEGIN 1
  SAY @326
  IF ~~ THEN REPLY @327 GOTO req
  IF ~~ THEN REPLY @328 GOTO sim
  // IF ~~ THEN REPLY @329 EXIT
END

IF ~~ THEN BEGIN sim
  SAY @330 IF ~~ THEN GOTO req
END

IF ~~ THEN BEGIN req
  SAY @331
  IF ~~ THEN REPLY @332 GOTO whytrust
  IF ~~ THEN REPLY @333 GOTO explain
END

IF ~~ THEN BEGIN whytrust
  SAY @334 = @335 = @336 = @337
  IF ~~ THEN REPLY @338 GOTO insane
  IF ~~ THEN REPLY @339 GOTO explain
END

IF ~~ THEN BEGIN insane
  SAY @340
  IF ~~ THEN GOTO explain
END

IF ~~ THEN BEGIN explain
  SAY @341 = @342 
  IF ~~ THEN REPLY @343 GOTO e1
  IF ~~ THEN REPLY @344 GOTO e2
END

IF ~~ THEN BEGIN e1 SAY @345 IF ~~ THEN GOTO e3 END
IF ~~ THEN BEGIN e2 SAY @346 IF ~~ THEN GOTO e3 END

IF ~~ THEN BEGIN e3
  SAY @347 = @348
  IF ~~ THEN GOTO main_quest
END

IF ~!NumTimesTalkedTo(0)~ THEN BEGIN main_quest
  SAY @349
  IF ~~ THEN REPLY @350 GOTO say_defenses
  IF ~~ THEN REPLY @35 GOTO say_people
  IF ~~ THEN REPLY @45 GOTO say_advice
  IF ~~ THEN REPLY @55 GOTO say_ritual
  IF ~~ THEN REPLY @65 GOTO say_betray
  IF ~~ THEN REPLY @75 GOTO say_reward
  IF ~~ THEN REPLY @85 GOTO say_whoare
  IF ~~ THEN REPLY @95 GOTO say_moretime
  IF ~~ THEN REPLY @100 GOTO g1
END

CHAIN2 RITPONT say_defenses
  @351 = @352 = @353 = @354 = @355 = @356 = @357
END RITPONT main_quest

CHAIN2 RITPONT say_people
  @36 = @37 = @38 = @39 = @40 = @41 = @42 = @43 = @44
END RITPONT main_quest

CHAIN2 RITPONT say_advice
  @46 = @47 = @48 = @49 = @50 = @51
END RITPONT main_quest

// explain the Ritual .. not!
IF ~~ THEN BEGIN say_ritual
  SAY @56 IF ~~ THEN REPLY @57 GOTO say_r2
END
IF ~~ THEN BEGIN say_r2
  SAY @58 IF ~~ THEN REPLY @59 GOTO say_r3
END
IF ~~ THEN BEGIN say_r3
  SAY @60 IF ~~ THEN REPLY @61 GOTO say_r4
END
IF ~~ THEN BEGIN say_r4
  SAY @62
  IF ~~ THEN GOTO main_quest
END

// cute little trick: we'll use a CHAIN to ourselves!
CHAIN2 RITPONT say_betray
  @66 = @67
END RITPONT main_quest

CHAIN2 RITPONT say_reward
  @76 = @77 = @78
END RITPONT main_quest

CHAIN2 RITPONT say_whoare
  @86 = @87 = @88 = @88 = @89
END RITPONT main_quest

IF ~~ THEN BEGIN say_moretime
  SAY @96 IF ~~ THEN EXIT
END

IF ~~ THEN BEGIN g1
  SAY @101
  IF ~~ THEN REPLY @102 GOTO trustme
  IF ~~ THEN REPLY @104 GOTO moretime2
END

IF ~~ THEN BEGIN trustme
  SAY @103 = @106
  IF ~~ THEN REPLY @107 GOTO accept
  IF ~~ THEN REPLY @104 GOTO moretime2
END

IF ~~ THEN BEGIN accept
  SAY @110 
  // ONE_DAY = 7200, so one hour = 300 time units
  IF ~~ THEN REPLY @112 DO ~SetGlobalTimer("RitualIsNowOver","GLOBAL",4500)~
        GOTO want_more_time
  IF ~~ THEN REPLY @113 DO ~SetGlobalTimer("RitualIsNowOver","GLOBAL",2100)~ 
        GOTO accept_fine
  IF ~~ THEN REPLY @114 DO ~SetGlobalTimer("RitualIsNowOver","GLOBAL",2100)~ 
        GOTO accept_heckle
END

IF ~~ THEN BEGIN want_more_time
  SAY @115
  IF ~~ THEN REPLY @116 GOTO accept_fine
END

IF ~~ THEN BEGIN accept_heckle
  SAY @117
  IF ~~ THEN GOTO accept_fine
END

IF ~~ THEN BEGIN accept_fine
  SAY @111
  IF ~~ THEN DO ~SetGlobal("RitualAccept","GLOBAL",1)~ EXIT
END

IF ~~ THEN BEGIN moretime2
  SAY @105 IF ~~ THEN GOTO main_quest
END

